### PR TITLE
Include flattened mappers in converter resolution

### PR DESCRIPTION
**Pull request by akiesler**
This ensures that flattened fields are correctly resolved for conversion. This mostly impacts extensions such as the AutoGeneratedTimestampRecordExtension which could not handle a flattend timestamp field.

https://github.com/aws/aws-sdk-java-v2/issues/3150

## Motivation and Context
This ensure that Extensions using the converterForAttribute method can find converters for all attributes.

https://github.com/aws/aws-sdk-java-v2/issues/3150

## Modifications
When looking for a converter also consider the flattened mappers for find the converter.

## Testing
Added and executed unit tests.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [x] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License
- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: .changes/next-release/bugfix-AmazonDynamoDB-965386f.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "type": "bugfix",
+    "category": "Amazon DynamoDB",
```
Comment (dagnir) [inline .changes/next-release/bugfix-AmazonDynamoDB-965386f.json:3]: Should change to "Amazon DynamoDB Enhanced Client"

**Review by dagnir — COMMENTED** [review]
Hi @akiesler thanks for the PR! The approach LGTM, just a small comment.

File: .changes/next-release/bugfix-AmazonDynamoDB-965386f.json (hunk 1 above)
Comment (akiesler) [reply .changes/next-release/bugfix-AmazonDynamoDB-965386f.json]: Updated and pushed

**Review by dagnir — APPROVED** [review]

**Comment by dagnir** [comment]
@all-contributors please add @akiesler for code

**Pull request merged**